components/calendar/calendar.js

```diff
@@ -811,11 +820,15 @@ class Calendar extends LocalizeCoreElement(RtlMixin(LitElement)) {
 	}
 
 	async _updateFocusDate(possibleFocusDate, latestPossibleFocusDate, allowDisabled) {
-		await this.updateComplete;
-		if (isDateInRange(possibleFocusDate, getDateFromISODate(this.minValue), getDateFromISODate(this.maxValue)) || allowDisabled) {
+		if ((!this.minValue && !this.maxValue)
+			|| (isDateInRange(possibleFocusDate, getDateFromISODate(this.minValue), getDateFromISODate(this.maxValue)) || allowDisabled)
```

Comment:
Similar to above these embedded function calls could be extracted onto their own lines